src/auth/auth.service.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  Injectable,
+  UnauthorizedException,
+  ConflictException,
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { User } from '../user/entities/user.entity';
+import { LoginUserDto } from './dto/login-auth.dto';
+import { ForgotPasswordRequestDto } from './dto/login-auth.dto';
+import { ResetPasswordRequestDto } from './dto/login-auth.dto';
+import { JwtService } from '@nestjs/jwt';
+import { authMessage, errorMessages, userMessages } from 'src/shared/constant/constant';
+import * as bcrypt from 'bcrypt';
+import { UserService } from 'src/user/user.service';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    @InjectRepository(User)
+    private userRepository: Repository<User>,
+    private jwtService: JwtService,
+    private readonly userService: UserService,
+  ) {}
+  private formatResponse(message: string, data: any) {
```

Comment:
this function is being repeated, let's move it to utilities and call it from there